### PR TITLE
Refactor identified enums in Reports

### PR DESCRIPTION
**Pull request by FaithOmbongi**

### CODE REVIEW

File: api-reference/beta/resources/statusdetails.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Describes the status of the provisioning event and the associated errors. It is
 
 | Property     | Type        | Description |
 |:-------------|:------------|:------------|
-|status|String|Possible values are: `success`, `failure`, `skipped`, `unknownFutureValue`. Inherited from statusBase.|
+|status|statusBase|Possible values are: `success`, `warning`, `failure`, `skipped`, `unknownFutureValue`. Inherited from statusBase.|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/statusdetails.md:22]: Verify BaseType

File: api-reference/beta/resources/credentialusagesummary.md [hunk 2]
```diff
@@ -25,9 +25,9 @@ Represents the current state of how many users in your organization are using se
 
 | Property     | Type        | Description |
 |:-------------|:------------|:------------|
-| authMethod | string | Represents the authentication method that the user used. Possible values are: `email`, `mobileSMS`, `mobileCall`, `officePhone`, `securityQuestion` (only used for self-service password reset), `appNotification`, `appCode`, and  `alternateMobileCall` (only supported for registration). |
+| authMethod | usageAuthMethod | Represents the authentication method that the user used. Possible values are:<ul><li>`email`</li><li> `mobileSMS`</li><li> `mobileCall`</li><li>`officePhone`</li><li>`securityQuestion` (only used for self-service password reset)</li><li>`appNotification`</li><li>`appCode`</li><li>`alternateMobileCall` (supported only in registration).</li><li>`fido`</li><li>`appPassword`</li><li>`unknownFutureValue`</li></ul> |
```
Comment (davidmu1) [inline api-reference/beta/resources/credentialusagesummary.md:28]: Even though the list is very long, let's not add lists to the property descriptions. It makes the pages much longer on pages where there could be many.

File: api-reference/beta/resources/credentialusagesummary.md [hunk 3]
```diff
@@ -25,9 +25,9 @@ Represents the current state of how many users in your organization are using se
...
 | failureActivityCount | Int64 | Provides the count of failed resets or registration data. |
-| feature | string | Defines the feature to report. Possible values are: `registration` and `reset`. |
+| feature | featureType | Defines the feature to report. Possible values are: `registration`, `reset` `unknownFutureValue`. |
```
Comment (davidmu1) [inline api-reference/beta/resources/credentialusagesummary.md:30]:
```suggestion
| feature | featureType | Defines the feature to report. Possible values are: `registration`, `reset`, `unknownFutureValue`. |
```

File: api-reference/beta/resources/directoryaudit.md [hunk 4]
```diff
@@ -34,8 +34,8 @@ Represents the directory audit items and its collection.
...
-| resultReason        | String                                              | Indicates the reason for failure if the result is "Failure" or "timeout".                                                                                                                                                                                              |
+| result              | operationResult                                              | Indicates the result of the activity. Possible values are: `success`, `failure`, `timeout`, `unknownFutureValue`.                                                                                                                                                       |
+| resultReason        | String                                              | Indicates the reason for failure if the result is "failure" or "timeout".                                                                                                                                                                                              |
```
Comment (davidmu1) [inline api-reference/beta/resources/directoryaudit.md:38]:
```suggestion
| resultReason        | String                                              | Indicates the reason for failure. Possible values: `failure` or `timeout`. |
```

File: api-reference/beta/resources/keyvaluepair.md [hunk 5]
```diff
@@ -19,8 +19,8 @@ Key value pair for action parameters.
 
 | Property     | Type        | Description |
 |:-------------|:------------|:------------|
-|name|String|Name for this key-value pair|
-|value|String|Value for this key-value pair|
+|name|String|Name for this key-value pair. <br><br>Possible names are: `AdditionalWSFedEndpointCheckResult`,  `AllowedAuthenticationClassReferencesCheckResult`, `AlwaysRequireAuthenticationCheckResult`,   `AutoUpdateEnabledCheckResult`, `ClaimsProviderNameCheckResult`, `EncryptClaimsCheckResult`,  `EncryptedNameIdRequiredCheckResult`, `MonitoringEnabledCheckResult`,`NotBeforeSkewCheckResult`,  `RequestMFAFromClaimsProvidersCheckResult`, `SignedSamlRequestsRequiredCheckResult`, `AdditionalAuthenticationRulesCheckResult`, `TokenLifetimeCheckResult`,  `DelegationAuthorizationRulesCheckResult`, `IssuanceAuthorizationRulesCheckResult`, `IssuanceTransformRulesCheckResult`.|
```
Comment (davidmu1) [inline api-reference/beta/resources/keyvaluepair.md:22]: Please remove the breaks.

File: api-reference/beta/resources/usercredentialusagedetails.md [hunk 6]
```diff
@@ -25,10 +25,10 @@ Represents the self-service password reset usage for a given tenant. Details inc
 
 | Property     | Type        | Description |
 |:-------------|:------------|:------------|
-| authMethod | string | Represents the authentication method that the user used. Possible values are: `email`, `mobileSMS`, `mobileCall`, `officePhone`, `securityQuestion` (only used for self-service password reset), `appNotification`, `appCode`, and `alternateMobileCall` (supported only in registration). |
+| authMethod | usageAuthMethod | Represents the authentication method that the user used. Possible values are:<ul><li>`email`</li><li> `mobileSMS`</li><li> `mobileCall`</li><li>`officePhone`</li><li>`securityQuestion` (only used for self-service password reset)</li><li>`appNotification`</li><li>`appCode`</li><li>`alternateMobileCall` (supported only in registration).</li><li>`fido`</li><li>`appPassword`</li><li>`unknownFutureValue`</li></ul> |
```
Comment (davidmu1) [inline api-reference/beta/resources/usercredentialusagedetails.md:28]: Remove list elements.

File: api-reference/v1.0/resources/appliedconditionalaccesspolicy.md [hunk 7]
```diff
@@ -21,7 +21,7 @@ Indicates the attributes related to applied conditional access policy or policie
 |enforcedGrantControls|String collection|Refers to the grant controls enforced by the conditional access policy (example: “Require multi-factor authentication”).|
 |enforcedSessionControls|String collection|Refers to the session controls enforced by the conditional access policy (example: “Require app enforced controls”).|
 |id|String|Unique GUID of the conditional access policy.|
-|result|String| Indicates the result of the CA policy that was triggered. Possible values are:<br/>`success`<br/>`failure`<br/>`notApplied` - Policy isn't applied because policy conditions were not met.<br/>`notEnabled` - This is due to the policy in disabled state.|
+|result|appliedConditionalAccessPolicyResult| Indicates the result of the CA policy that was triggered. Possible values are:<br/>`success`<br/>`failure`<br/>`notApplied` - Policy isn't applied because policy conditions were not met.<br/>`notEnabled` - This is due to the policy in disabled state.<br/>`unknown`<br/>`unknownFutureValue`<br/>`reportOnlySuccess`<br/>`reportOnlyFailure`<br/>`reportOnlyNotApplied`<br/>`reportOnlyInterrupted`|
```
Comment (davidmu1) [inline api-reference/v1.0/resources/appliedconditionalaccesspolicy.md:24]: Remove breaks.

**Review by davidmu1 — APPROVED** [review]

**Pull request merged**